exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,285 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
+			Tags:  typeIsHistogramTags,
+		})
+}
+
+func TestHistogramReportingError(t *testing.T) {
+	report := newHistogramReporting(zap.NewNop())
+	sender := &mockGaugeSender{errorOnSend: true}
+	var errs []error
+
+	report.Report(sender, &errs)
+
+	assert.NotEmpty(t, errs)
+}
+
+func TestCumulativeHistogramDataPointConsumer(t *testing.T) {
+	verifyCumulativeHistogramDataPointConsumer(t, false)
+}
+
+func TestCumulativeHistogramDataPointConsumerError(t *testing.T) {
+	verifyCumulativeHistogramDataPointConsumer(t, true)
+}
+
+func TestCumulativeHistogramDataPointConsumerLeInUse(t *testing.T) {
+	metric := newMetric("a.metric", pdata.MetricDataTypeHistogram)
+	histogramDataPoint := pdata.NewHistogramDataPoint()
+	histogramDataPoint.SetBucketCounts([]uint64{4, 12})
+	histogramDataPoint.SetExplicitBounds([]float64{10.0})
+	setTags(map[string]interface{}{"le": 8}, histogramDataPoint.Attributes())
+	sender := &mockGaugeSender{}
+	report := newHistogramReporting(zap.NewNop())
+	consumer := newCumulativeHistogramDataPointConsumer(sender)
+	var errs []error
+
+	consumer.Consume(metric, histogramDataPoint, &errs, report)
+
+	assert.Empty(t, errs)
+	assert.Empty(t, sender.metrics)
+	assert.Equal(t, int64(1), report.LeTagInUse())
+	assert.Equal(t, int64(0), report.Malformed())
+}
+
+func TestCumulativeHistogramDataPointConsumerMissingBuckets(t *testing.T) {
+	metric := newMetric("a.metric", pdata.MetricDataTypeHistogram)
+	histogramDataPoint := pdata.NewHistogramDataPoint()
+	sender := &mockGaugeSender{}
+	report := newHistogramReporting(zap.NewNop())
+	consumer := newCumulativeHistogramDataPointConsumer(sender)
+	var errs []error
+
+	consumer.Consume(metric, histogramDataPoint, &errs, report)
+
+	assert.Empty(t, errs)
+	assert.Empty(t, sender.metrics)
+	assert.Equal(t, int64(1), report.Malformed())
+	assert.Equal(t, int64(0), report.LeTagInUse())
+}
+
+// Creates a histogram metric with len(countAttributeForEachDataPoint)
+// datapoints. name is the name of the histogram metric; temporality
+// is the temporality of the histogram metric;
+// countAttributeForEachDataPoint contains the count attribute for each
+// data point.
+func newHistogramMetricWithDataPoints(
+	name string,
+	temporality pdata.MetricAggregationTemporality,
+	countAttributeForEachDataPoint []uint64,
+) pdata.Metric {
+	result := newMetric(name, pdata.MetricDataTypeHistogram)
+	histogram := result.Histogram()
+	histogram.SetAggregationTemporality(temporality)
+	histogram.DataPoints().EnsureCapacity(len(countAttributeForEachDataPoint))
+	for _, count := range countAttributeForEachDataPoint {
+		histogram.DataPoints().AppendEmpty().SetCount(count)
+	}
+	return result
+}
+
+func verifyCumulativeHistogramDataPointConsumer(t *testing.T, errorOnSend bool) {
+	metric := newMetric("a.metric", pdata.MetricDataTypeHistogram)
+	histogramDataPoint := pdata.NewHistogramDataPoint()
+	histogramDataPoint.SetBucketCounts([]uint64{2, 3, 5, 7})
+	histogramDataPoint.SetExplicitBounds([]float64{2.0, 5.0, 10.0})
```

Comment:
Can we set the Explicit Bounds first, following by Bucket Counts? I think it has more meaning to know what they bounds are first, before adding datapoints that fall into those buckets. Also, a quick explanation of what this test data means, like:
```
/*
  Create bounds of -Inf to <= 2.0, > 2.0 to <= 5.0, >5.0 to <= 10.0, > 10.0 to +Inf
  Set 2 data points in first bucket, 3 in second, 5 in third, 7 in fourth
*/
```